src/services/TestResultsService.js

```diff
@@ -16,64 +17,64 @@ class TestResultsService {
+      if (filters.fromDateTime && filters.toDateTime) {
+        if (!GetTestResults.validateDates(filters.fromDateTime, filters.toDateTime)) {
+          return Promise.reject(new HTTPError(400, 'Bad request'))
         }
-        let testResults = data.Items
-        if (testResults !== null) {
-          testResults = this.filterTestResultsByStatus(testResults, status)
-          testResults = this.filterTestResultByDate(testResults, fromDateTime, toDateTime)
-          if (testResults.length === 0) {
-            throw new HTTPError(404, 'No resources match the search criteria')
+      }
+      if (filters.vin) {
+        return this.testResultsDAO.getByVin(filters.vin).then(response => {
+          return this.checkDAOResponse(response, filters)
+        }).catch(error => {
+          if (!(error instanceof HTTPError)) {
+            console.error(error)
+            error = new HTTPError(500, 'Internal Server Error')
           }
-        }
-        testResults = this.removeTestResultId(testResults)
-        testResults = testResults.map((testResult) => this.removeVehicleClassification(testResult))
-        return testResults
-      })
-      .catch(error => {
-        if (!(error instanceof HTTPError)) {
-          console.error(error)
-          error = new HTTPError(500, 'Internal Server Error')
-        }
-        throw error
-      })
-  }
-
-  removeTestResultId (testResults) {
-    if (testResults) {
-      for (let i = 0; i < testResults.length; i++) { delete testResults[i].testResultId }
+          throw error
+        })
+      } else if (filters.testerStaffId) {
+        return this.testResultsDAO.getByTesterStaffId(filters.testerStaffId).then(data => {
+          return this.checkDAOResponse(data, filters)
+        }).catch(error => {
+          if (!(error instanceof HTTPError)) {
+            console.error(error)
+            error = new HTTPError(500, 'Internal Server Error')
+          }
+          throw error
+        })
+      } else {
+        return Promise.reject(new HTTPError(400, 'Bad request'))
+      }
+    } else {
+      return Promise.reject(new HTTPError(400, 'Bad request'))
     }
-    return testResults
   }
-
-  filterTestResultsByStatus (testResults, status) {
-    var filteredTestResults = testResults.filter(
-      function (testResult) { return testResult.testStatus === status }
-    )
-    return filteredTestResults
+  checkTestResults (data) {
+    if (data) {
+      if (!data.Count) {
+        throw new HTTPError(404, 'No resources match the search criteria')
+      }
+    }
+    return data.Items
   }
 
-  filterTestResultByDate (testResults, fromDateTime, toDateTime) {
-    for (let i = 0; i < testResults.length; i++) {
-      testResults[i].testTypes = testResults[i].testTypes.filter(
-        function (testType) {
-          return (!(dateFns.isAfter(testType.createdAt, toDateTime) || dateFns.isBefore(testType.createdAt, fromDateTime)))
-        })
+  checkDAOResponse (data, filters) {
```

Comment:
I think this function should be renamed to something like `filterTestResults` / `applyTestResultsFilters`

Reply:
done